cartoframes/data/enrichment/enrichment_service.py

```diff
@@ -120,29 +133,63 @@ def __get_tables_and_variables(variables):
     else:
         raise EnrichmentException('Variable(s) to enrich should be an instance of Series or DataFrame')
 
-    table_to_variables = __process_enrichment_variables(variables_id)
-    table_data_enrichment = list(table_to_variables.keys()).pop()
-    table_geo_enrichment = __get_name_geotable_from_datatable(table_data_enrichment)
-    variables_list = list(table_to_variables.values()).pop()
+    table_to_geotable, table_to_variables, table_to_project, table_to_dataset =\
+        __process_enrichment_variables(variables_id, user_dataset)
 
-    return table_data_enrichment, table_geo_enrichment, variables_list
+    return table_to_geotable, table_to_variables, table_to_project, table_to_dataset
 
 
-def __process_enrichment_variables(variables):
+def __process_enrichment_variables(variables, user_dataset):
+    table_to_geotable = dict()
     table_to_variables = defaultdict(list)
+    table_to_project = dict()
+    table_to_dataset = dict()
 
     for variable in variables:
         variable_split = variable.split('.')
-        table, variable = variable_split[-2], variable_split[-1]
+        project_part, dataset_part, table_part, variable_part = variable_split
+
+        if project_part != _PUBLIC_PROJECT:
+            table_part = '{dataset}_{table}'.format(dataset=dataset_part,
+                                                    table=table_part,
+                                                    user_dataset=user_dataset)
+
+        if table_part not in table_to_dataset:
+            if project_part != _PUBLIC_PROJECT:
+                table_to_dataset[table_part] = user_dataset
+            else:
+                table_to_dataset[table_part] = _PUBLIC_DATASET
 
-        table_to_variables[table].append(variable)
+        if table_part not in table_to_geotable:
+            geotable = __get_name_geotable_from_datatable(table_part)
 
-    return table_to_variables
+            if project_part != _PUBLIC_PROJECT:
+                geotable = '{dataset}_{geotable}'.format(dataset=dataset_part,
+                                                         geotable=geotable,
+                                                         user_dataset=user_dataset)
+
+            table_to_geotable[table_part] = geotable
+
+        if table_part not in table_to_project:
+            if project_part == _PUBLIC_PROJECT:
+                table_to_project[table_part] = _PUBLIC_PROJECT
+            else:
+                table_to_project[table_part] = _WORKING_PROJECT
+
+        table_to_variables[table_part].append(variable_part)
+
+    return table_to_geotable, table_to_variables, table_to_project, table_to_dataset
 
 
 def __get_name_geotable_from_datatable(datatable):
+
     datatable_split = datatable.split('_')
-    geo_information = datatable_split[2:5]
+
+    if len(datatable_split) == 8:
```

Comment:
I don't see any other solution right now, but this part smells :(

Reply:
Yes, I think we have to think this better in the future, it is because of the addition of provider name in the name of the table when licensing a table. So if you are using a licensed dataset or a public dataset the name structure change